TACC/Core-Portal
# task/TV3-141 & task/TV3-171 - Fix Navigation in Copy/Move Modal & Incorrect System Names

## Overview

This PR contains changes to breadcrumbs and the system name that is displayed in the breadcrumb. It contains the fix for a bug where the correct system was not being selected and the wrong system name was being shown in the breadcrumb and search placeholder

This PR also simplifies the navigation in the Data Listing modal when the user was in the home directory. This was done by removing the back button when they are in the home directory as that is the farthest the user should be allowed to go. This also includes a fix for the navigation breadcrumbs in the Show Path modal

## Related

* [TV3-141](https://jira.tacc.utexas.edu/browse/TV3-141)
* [TV3-171](https://jira.tacc.utexas.edu/browse/TV3-171)

## Changes


- Added `path` variable to `useSystemDisplayName`. Previously only the `system` and `scheme` parameters were used to select a system. With the new system object definition the systems **My Data (Frontera)** and **My Data (Frontera)** had the same system and scheme which made it difficult to select the correct system. With the new path variable an additional comparison is made to a systems `homeDir` value to get the correct system
- Created a new central utility hook called `fetchSelectedSystem` that retrieves the current active system. This prevents previous code duplication and ensures that the correct system is selected to display and can be easily reused in other components. This has a similar process as `useSystemDisplayName` for selecting the correct system
- If the user were to navigate to the root of a system, it will now show the name of the base system (e.g Frontera) instead of the assigned system name (i.e My Data (Frontera)). 
- Added a check to see if the path in the DataFilesListingModal is the `homeDir` or not and hide/show the back button accordingly
- Fixed breadcrumbs in the Show Path modal by only sending in the file path as the prop

## Testing

1. Go to Data Files and switch between the different systems and ensure correct system name is displayed
2. Go to Data Files and open the Copy or Move modal
3. Ensure back button is not visible when in the home directory
4. Click on Show Path for any file and ensure navigation breadcrumb is correct

## UI

**Before:**

Incorrect system name being displayed
<img width="1912" alt="Screen Shot 2023-05-18 at 2 11 57 PM" src="https://github.com/TACC/Core-Portal/assets/23081932/ce7ec584-3c3f-457e-a088-ed956d0f03aa">

System name showing at root directory of system
<img width="1912" alt="Screen Shot 2023-05-18 at 2 14 15 PM" src="https://github.com/TACC/Core-Portal/assets/23081932/486b5df1-e015-4b5f-8819-6f58d2a87e14">

Unnecessary back button in copy modal
<img width="1114" alt="Screen Shot 2023-04-24 at 2 51 12 PM" src="https://user-images.githubusercontent.com/23081932/234101232-4e4a0636-b9c2-465d-92a1-573d99f8fd81.png">

Incorrect breadcrumbs in Show Path modal
<img width="473" alt="Screen Shot 2023-04-25 at 9 55 48 AM" src="https://user-images.githubusercontent.com/23081932/234319170-1e5a0124-4861-4f06-b6b7-c5a19a524f9d.png">


**After:**

Correct system name
<img width="1912" alt="Screen Shot 2023-05-18 at 2 13 30 PM" src="https://github.com/TACC/Core-Portal/assets/23081932/a5ac72d3-5c22-4a70-a062-cea8ddba394a">

Base system name being show in root directory
<img width="1912" alt="Screen Shot 2023-05-18 at 2 14 24 PM" src="https://github.com/TACC/Core-Portal/assets/23081932/cf0a4517-4556-4211-a8e0-b9a0b4594495">

Removed back button from modal
<img width="1114" alt="Screen Shot 2023-04-24 at 2 51 03 PM" src="https://user-images.githubusercontent.com/23081932/234101290-9f29ec0b-879e-4697-a8fc-6aff99c6384b.png">

Correct path showing in Show Path modal
<img width="473" alt="Screen Shot 2023-04-25 at 9 54 29 AM" src="https://user-images.githubusercontent.com/23081932/234319241-ec800ed8-9e9f-4d4f-85be-c9e432799a0f.png">

## Notes
